src/axios-har-tracker.ts

```diff
@@ -1,112 +1,126 @@
-const axios = require('axios').default;
-import { writeFileSync } from 'fs';
+import { AxiosStatic, AxiosRequestConfig, AxiosPromise } from 'axios';
 import * as cookie from 'cookie';
 
-const date = new Date();
-const startDate = date.toISOString();
+export class AxiosHarTracker {
 
-let generatedHar = {
-  log: {
-    version: '1.2',
-    creator: {
-      name: 'axios-tracker',
-      version: '1.0.0'
-    },
-    pages: [],
-    entries: []
-  }
-};
+  private axios: AxiosStatic;
+  private generatedHar;
+  private newEntry;
 
-let newEntry = {
-  request: {},
-  response: {},
-  startedDateTime: startDate,
-  time: -1,
-  cache: {},
-  timings: {
-    blocked: -1,
-    dns: -1,
-    ssl: -1,
-    connect: -1,
-    send: 10,
-    wait: 10,
-    receive: 10,
-    _blocked_queueing: -1
-  }
-};
+  constructor(requestModule: AxiosStatic) {
+    this.axios = requestModule;
 
-async function axiosTracker(call) {
-  axios.interceptors.request.use(
-    async config => {
-      config.validateStatus = function () {
-        return true;
-      };
-      config.headers['request-startTime'] = process.hrtime();
-      const fullCookie = JSON.stringify(config.headers['Cookie']);
-      const version = config.httpVersion === undefined ? 'HTTP/1.1' : 'HTTP/' + config.httpVersion;
+    this.generatedHar = {
+      log: {
+        version: '1.2',
+        creator: {
+          name: 'axios-tracker',
+          version: '1.0.0'
+        },
+        pages: [],
+        entries: []
+      }
+    };
 
-      newEntry.request = {
-        method: config.method,
-        url: config.url,
-        httpVersion: version,
-        cookies: getCookies(fullCookie),
-        headers: [],
-        queryString: getParams(config.params),
-        headersSize: -1,
-        bodySize: -1
-      };
-      return config;
-    },
-    error => {
-      return Promise.reject(error);
-    }
-  );
+    this.newEntry = {
+      request: {},
+      response: {},
+      startedDateTime: new Date().toISOString(),
```

Comment:
startedDateTime is set in the constructor so it will be applied only once, during creation of an instance. Date assignment has to happen in the interceptor function

Reply:
Please be aware that there are two kinds of startedDateTime - 1st one in newEntry, 2nd on in interceptors

Reply:
Ok, in that case, it's not needed when newEntry is initialized, it should be setup only in the interceptor